packages/http-client/src/http-client.ts

```diff
@@ -124,6 +125,30 @@ export function execute<ReturnT>(executeFn: ExecuteHttpRequestFn<ReturnT>) {
   };
 }
 
+/**
+ * Build an [[HttpRequestConfigWithOrigin]] from a given [[HttpRequestConfigWithOrigin]] or [[HttpRequestConfig]]
+ * @param requestConfig - The given [[HttpRequestConfigWithOrigin]] or [[HttpRequestConfig]]
+ * @returns The resulting [[HttpRequestConfigWithOrigin]]
+ * @internal
+ */
+export function buildHttpRequestConfigWithOrigin(
+  requestConfig: HttpRequestConfigWithOrigin | HttpRequestConfig
+): HttpRequestConfigWithOrigin {
+  if (isHttpRequestConfigWithOrigin(requestConfig)) {
+    return requestConfig;
+  }
+  let ret: HttpRequestConfigWithOrigin = {
+    ...(requestConfig as HttpRequestConfigBase)
+  };
+  ret = requestConfig.headers
+    ? { ...ret, headers: { requestConfig: {}, custom: requestConfig.headers } }
+    : ret;
+  ret = requestConfig.params
+    ? { ...ret, params: { requestConfig: {}, custom: requestConfig.params } }
+    : ret;
+  return ret;
```

Comment:
The only difference is that, when `headers` is `undefined`:
- my code will return an object without the `headers` key
- your version will be `{ headers: { requestConfig: {}, custom: undefined} }`
I considered the second one, but feel a bit inconsistent.

Reply:
Why do you think this is inconsistent? The custom keyword will always be there, therefore it seems even more consistent to me ;)
I think the shorter implementation reads much better and has less code duplication. It should technically not be a problem and the only difference.
If you think that this can cause issues, maybe consider this:

```ts
return {
    ...requestConfig,
    headers: { requestConfig: {}, ...requestConfig.headers && { custom: requestConfig.headers } },
    params: { requestConfig: {}, ...requestConfig.params && { custom: requestConfig.params } }
  };
```

But honestly, to me this seems like an unnecessary complication.